api/server/server.go

```diff
@@ -1123,6 +1126,142 @@ func postContainersCopy(eng *engine.Engine, version version.Version, w http.Resp
 	return nil
 }
 
+// handleArchiveError is a convenience function for the following Copy
+// handlers which makes a JSON error message but leaves writing the
+// header and response to the package's common error handler.
+func handleArchiveError(w http.ResponseWriter, err error) error {
```

Comment:
As with the previous PR, I'd really prefer to keep all http error handing in one spot so that when we do https://github.com/docker/docker/pull/9509 life will be easier.  Any chance of moving this into httpError() and remove the http-specific logic/calls from funcs like getContainersStatPath() ?  IOW, just return an error and let httpError() deal with the http stuff.